mautic/mautic
# [UX] Disable "Visible on quick add" when creating a Custom Field for Company object

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #14178 



## Description
This PR aims to resolve #14178.
> But when you are creating a custom field for a company, you do not have the quick add functionality, but this toggle keeps visible and interactive anyway. That's problematic and confusing for the user because no matter what you choose, absolutely nothing will happen.

> To fix this, the `<div class="toggle">` element for the field "Visible on quick add" must receive the class toggle--disabled when Object is set to Company, so the final element becomes `<div class="toggle toggle--disabled">` to block user interaction and display the not-allowed cursor.

In this PR 
- Created js function `toggleIsShortVisibleVisibility` to toggle  `toggle--disabled` class on the isShortVisible container
- Added call inside `leadfieldOnLoad` and when the leadtype for field object changes






---
### 📋 Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Open Custom Fields
3. Create a new custom field
4. Change Object to company and see that "Visible on quick add" is disabled
6. Fill the required fields and hit save
7. The page should refresh and the field should still be disabled.
8. Repeat step 3 & 4
9. Change  Object back to contact and the field should be enabled again
10. Fill the required fields and hit save
11. The page should refresh and the field should still be enabled.
